### DISCUSSION TITLE
[WASM] ClientWebSocket request headers hang web app

### DISCUSSION THREAD

**Discussion by emceelovin** [unoplatform/uno#7808] [Q&A]
### Current behavior

Setting request headers via ClientWebSocket.Options.SetRequestHeader causes WASM applicaton to freeze at the call to SetRequestHeader. Commenting out that line allows the program to function 100% correctly

### Expected behavior

Should be allowed to set request headers

### Workaround

Comment call to SetRequestHeader

**Comment by jeromelaban** [comment]
Thanks for the report. Could you create a repro app which shows the issue?

**Reply by emceelovin** [reply]
Absolutely! The connection code I have is very simple. It doesn't matter if this is a blank project or not. Hope this helps!

```cs
clientSocket = new ClientWebSocket();

try
{
    // this works on wasm
    //if (null != Headers)
    //    foreach (var header in Headers)
    //        clientSocket.Options.SetRequestHeader(header.Key, header.Value);

    // works in uwp
    // works on android
    // causes wasm app to hang
    if (null != Headers)
        foreach (var header in Headers)
            clientSocket.Options.SetRequestHeader(header.Key, header.Value);

    await clientSocket.ConnectAsync(_uri, connectToken);
}

...
````

**Reply by jeromelaban** [reply]
Thanks for the repro. In this case, it'll be an issue to open in [dotnet runtime](https://github.com/dotnet/runtime), considering the fact that you are using net6.0 in the targetframework of your wasm head.

**Reply by emceelovin** [reply]
This isn't a dotnet 6 uno project template. I'm on dotnet 5 almost certain. As the comments suggest also, it works on uwp and android. which are on the dotnet and mono runtimes.

**Reply by jeromelaban** [reply]
Thanks, if you're running uno.wasm.bootstrap 3.0 or later, it's running on .NET 6 anyways, and this still means it's a .NET runtime issue.

**Reply by emceelovin** [reply]
that makes sense. i figured it was an issue between dotnet and wasm, but i can't get any other error out of the IDE. I'll see what the chrome debugger console says and paste that in here.. Give me a second.
